scripts/generate_big_data_network.py

```diff
@@ -0,0 +1,358 @@
+import os
+import argparse
+import numpy as np
+import random
+import json
+import sys
+
+import astroplan
+from astroplan import Observer
+from astroplan import download_IERS_A
+from datetime import datetime
+from datetime import timedelta
+from astropy.time import Time
+from astropy.io import fits
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy.coordinates.angles import Angle
+from astropy.coordinates import EarthLocation
+from collections import defaultdict
+
+from pocs.utils.google.storage import PanStorage
+
+
+#############################################################################
+# This class is a simulator that is designed to simulate larger quantities  #
+# of data, more accurate in the levels of data products than                #
+# generate_data_network.py (which has more realistic star data). Use this   #
+# to simulate large quantities of data to test the pipeline. Should         #
+# eventually be merged with generate_data_network.py.                       #
+#############################################################################
+
+class DataGenerator(object):
+    """Class to generate a network of simulated PANOPTES images and light curves
+
+    Now just a wrapper class, should be updated to abstract out other objects eventually
+    """
+
+    def __init__(self, storage=None):
+        if not storage:
+            storage = PanStorage(bucket_name='panoptes-simulated-data')
+        self.storage = storage
+        self.cameras = defaultdict(list)
+        self.star_dict = {}
+        self.unit_dict = {}
+        download_IERS_A()
+
+    def generate_network(self, num_units, start_date, end_date):
+        """Generate simulated data from a network of PANOPTES units.
+
+        :param num_units: the number of units to simulate
+        :param start_date: the start date of observations
+        :param end_date: the end date of observations
+        """
+        start_date = datetime.strptime(start_date, "%Y-%m-%d")
+        end_date = datetime.strptime(end_date, "%Y-%m-%d")
+        num_nights = (end_date - start_date).days + 1
+        if end_date < start_date:
+            raise ValueError('End date must be after start date.')
+
+        units = self.get_current_network()
```

Comment:
num_units is used to define the number of units if there is currently no data in the storage bucket (line 64). if there already is a network in the cloud, the units from there are taken and num_units is ignored. it's fine if this functionality changes based on how this should be used, for instance if we want to run it locally (see comment below)